driver-core/src/main/java/com/datastax/driver/core/DefaultResultSetFuture.java

```diff
@@ -91,7 +91,7 @@ public void onSet(Connection connection, Message.Response response, ExecutionInf
                                         else
                                             keyspace.removeTable(scc.columnFamily);
                                     }
-                                    this.setResult(rs);
+                                    session.cluster.manager.waitForSchemaAgreementAndSignal(connection, this, rs);
```

Comment:
For an earlier ticket I removed the schema refresh for DROP statements, but as a side-effect this also removed the schema agreement check. I think it's still useful and consistent to do it.